src/handlers.ts

```diff
@@ -304,6 +304,101 @@ export async function replyAction(
     });
 }
 
+export function createUnfollowActionHandler(accountService: AccountService) {
+    return async function unfollowAction(
+        ctx: Context<{ Variables: HonoContextVariables }>,
+    ) {
+        const handle = ctx.req.param('handle');
+        const apCtx = fedify.createContext(ctx.req.raw as Request, {
+            db: ctx.get('db'),
+            globaldb: ctx.get('globaldb'),
+            logger: ctx.get('logger'),
+        });
+
+        const actorToUnfollow = await lookupObject(apCtx, handle);
+
+        if (!isActor(actorToUnfollow)) {
+            return new Response(null, {
+                status: 404,
+            });
+        }
+
+        const account = await accountService.getDefaultAccountForSite(
+            ctx.get('site'),
+        );
+
+        if (actorToUnfollow.id!.href === account.ap_id) {
+            return new Response(null, {
+                status: 400,
+            });
+        }
+
+        let unfolloweeAccount = await accountService.getAccountByApId(
+            actorToUnfollow.id!.href,
+        );
+
+        // TODO I think we can exit early here - there is obviously no follow relation if there is no account
+        if (!unfolloweeAccount) {
+            unfolloweeAccount = await accountService.createExternalAccount(
+                await mapActorToExternalAccountData(actorToUnfollow),
+            );
+        }
+
+        const isFollowing = await accountService.checkIfAccountIsFollowing(
+            account,
+            unfolloweeAccount,
+        );
+
+        if (!isFollowing) {
+            return new Response(null, {
+                status: 409,
+            });
+        }
+
+        // Need to get the follow
+        const unfollowId = apCtx.getObjectUri(Undo, {
+            id: uuidv4(),
+        });
+
+        const follow = new Follow({
+            id: null,
```

Comment:
🤔 This is what you was talking about with it being tricky to grab the original follow activity?

I assume this is safe to do, i guess other implementations are just going to grab the object anyways rather than try match IDs of activities up (as that would mean they also have to store the original activity)

Reply:
Yeah this works for our implementation, as well as mastodons, and we can address full compatibility in future!